main.rb

```diff
@@ -0,0 +1,185 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    while i < length
+      temp.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    while i < length
+      return false unless yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    while i < length
+      return true if yield(self[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?
+    i = 0
+    while i < length
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
```

Comment:
* [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1, true, 'hi', []].my_none? != [1, true, 'hi', []].none?
[1, false, 'hi', []].my_none? != [1, false, 'hi', []].none?
[1,2,3,4,5].my_none?(Integer) != [1,2,3,4,5].none?(Integer)
["word",2,3,4,5].my_none?(Integer) != ["word",2,3,4,5].none?(Integer)
['saheed', 'oladele', 'suretrust'].my_none?(/d/) != ['saheed', 'oladele', 'suretrust'].none?(/d/)
[1,2,3,4,5].my_none?(3) != [1,2,3,4,5].none?(3)
```